src/css-image-value.js

```diff
@@ -14,48 +14,47 @@
 
 (function(internal, scope) {
 
-  (function() {
-    function CSSImageValue() {
-      throw new TypeError("Can\'t instantiate CSSImageValue");
-    }
-    internal.inherit(CSSImageValue, CSSResourceValue);
+  function onLoad() {
+    this.state = "loaded";
+    this.intrinsicWidth = this._image.naturalWidth;
+    this.intrinsicHeight = this._image.naturalHeight;
+    if (this.intrinsicHeight != 0)
+      this.intrinsicRatio = this.intrinsicWidth / this.intrinsicHeight;
+  }
 
-    scope.CSSImageValue = CSSImageValue;
-  })();
+  function onError() {
+    this.state = "error";
+  }
 
-  var CSSImageValue = function(image) {
-    if (!(image instanceof Image)) {
-      throw new TypeError("image must be an Image object");
-    }
+  function onProgress() {
+    this.state = "loading";
+  }
 
-    function onLoad() {
-      this.state = "loaded";
-      this.intrinsicWidth = this._image.naturalWidth;
-      this.intrinsicHeight = this._image.naturalHeight;
-      if (this.intrinsicHeight != 0)
-        this.intrinsicRatio = this.intrinsicWidth / this.intrinsicHeight;
-    }
+  var CSSImageValue = function() {
+    throw new TypeError('CSSImageValue cannot be instantiated');
+  }
+  internal.inherit(CSSImageValue, CSSResourceValue);
 
-    function onError() {
-      this.state = "error";
-    }
+  scope.CSSImageValue = CSSImageValue;
 
-    function onProgress() {
-      this.state = "loading";
+  (function() {
+    function CSSImageValue(image) {
+      if (!(image instanceof Image)) {
+        throw new TypeError("image must be an Image object");
+      }
+
+      this._image = image;
+      this.state = "unloaded";
+      this.intrinsicWidth = null;
+      this.intrinsicHeight = null;
+      this.intrinsicRatio = null;
+      this._image.onload = onLoad.bind(this);
```

Comment:
This is unrelated to this patch, but I don't think this does what you want. If the `Image` is already loaded, then this will never be called.